test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -68,11 +70,32 @@
       app {:jruby-puppet
            {:max-active-instances num-jrubies}
            :versioned-code
-           {:code-id-command (script-path "warn_echo_and_error")}}
+           {:code-id-command (script-path "warn_echo_and_error")
+            :code-content-command (script-path "echo")}}
       (let [catalog (testutils/get-catalog)]
         (is (nil? (get catalog "code_id")))
         (is (logged? #"Non-zero exit code returned while running" :error))
-        (is (logged? #"Executed an external process which logged to STDERR: production" :warn)))))))
+        (is (logged? #"Executed an external process which logged to STDERR: production" :warn))))))
+  (testing "code id is not added and 500 is returned if code-content-command is not set"
+    (logging/with-test-logging
+     (bootstrap/with-puppetserver-running
+      app {:jruby-puppet
+           {:max-active-instances num-jrubies}
+           :versioned-code
+           {:code-id-command (script-path "echo")}}
+      (let [response (testutils/http-get "puppet/v3/catalog/localhost?environment=production")]
+        (is (= 500 (:status response)))
+        (is (logged? #"Error 500 on SERVER"))))))
```

Comment:
Ah, yes, that case should be an error.

Also, while we're on the subject ...

How do we feel about HTTP 500 for these cases?  I don't love it, as it always seems to indicate something intrinsically wrong with the server (i.e. "my fault" (as a developer)) as opposed to a misconfiguration.  I looked into this a while ago and didn't come up with a great alternative, the best one being 409 CONFLICT.